### PR TITLE
Added batch_normalization in the numpy backend.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
### Summary

Added the op `batch_normalization` in the numpy backend.

### PR Overview

* Added the function in the numpy backend
* Added the numpy implementation in the docs
* Added tests.

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/backend/backend_test.py [hunk 1]
```diff
@@ -1655,6 +1655,34 @@ def test_bias_add(self):
...
+                        reason='Theano behaves differently '
+                               'because of the broadcast.')
+    @pytest.mark.parametrize('axis', [1, -1])
+    @pytest.mark.parametrize('x_shape', [(3, 2, 4, 5), (3, 2, 4)])
+    def test_batch_normalization(self, axis, x_shape):
+        other_shape = [1] * len(x_shape)
+        other_shape[axis] = x_shape[axis]
+        other_shape = tuple(other_shape)
+        x_np = np.random.random(x_shape)
+        mean_np = np.random.random(other_shape)
```
Comment (taehoonlee) [inline tests/keras/backend/backend_test.py]: How about the following?
```
mean, var, beta, gamma = [np.random.random(other_shape).astype(np.float32)
                          for _ in range(4)]
```

File: tests/keras/backend/backend_test.py [hunk 2]
```diff
@@ -1655,6 +1655,34 @@ def test_bias_add(self):
...
+        var_np = np.random.random(other_shape)
+        beta_np = np.random.random(other_shape)
+        gamma_np = np.random.random(other_shape)
+        output_tensors = []
+        output_arrays = []
+        for k in WITH_NP:
+            x = k.variable(x_np)
+            mean = k.variable(mean_np)
```
Comment (taehoonlee) [inline tests/keras/backend/backend_test.py]: How about the following?
```python
mean_k, var_k, beta_k, gamma_k = [k.variable(v)
                                  for v in [x, mean, var, beta, gamma]]
```

File: tests/keras/backend/backend_test.py (hunk 1 above)
Comment (gabrieldemarmiesse) [reply tests/keras/backend/backend_test.py]: I suppose it's a style matter, I'd prefer avoid breaking lines unless necessary. I actually wanted to use #11265 for this test of batchnorm.

**Comment by gabrieldemarmiesse** [comment]
@taehoonlee ping. PR ready for review.
